### PR TITLE
Support application/x-www-form-urlencoded

### LINKED ISSUE #78: Support application/x-www-form-urlencoded
Handle `content-type`: `application/x-www-form-urlencoded` with https://github.com/nox/serde_urlencoded as suggest by @carllerche on Gitter.

### PR DESCRIPTION
**Pull request by EloD10**
Fix #78 
It should be good.

Whatever thanks a lot to @carllerche, you helped and encouraged me a lot!

### CODE REVIEW

File: src/extract/serde.rs [hunk 1]
```diff
@@ -140,16 +145,20 @@ where T: DeserializeOwned,
...
-                            Some(Error::internal_error())
+                    
+                    if self.is_json == true {
+                        ::serde_json::from_slice(&res[..])
+                            .map_err(|_| {
+                                // TODO: Handle error better
+                                Some(Error::internal_error())
+                        })
+                    } else {
+                        serde_urlencoded::from_bytes(&res[..])
```
Comment (bryanburgers) [inline src/extract/serde.rs]: Should this be `::serde_urlencoded` to match `::serde_json` above?

File: src/extract/serde.rs [hunk 2]
```diff
@@ -94,21 +95,25 @@ where T: DeserializeOwned,
...
 
-                SerdeFuture { state }
+                            SerdeFuture { state, is_json: true }  
+                        }
+                        b"application/x-www-form-urlencoded" => {
+                            let state = State::Body(body.collect());
+                            
+                            SerdeFuture { state, is_json: false }   
+                        }
+                        _ => panic!("Content type doesn't know")
```
Comment (bryanburgers) [inline src/extract/serde.rs]: "Unknown content type" flows a little bit better.

File: src/extract/serde.rs [hunk 3]
```diff
@@ -94,21 +95,25 @@ where T: DeserializeOwned,
...
-                            content_type.as_bytes().to_ascii_lowercase() ==
-                                b"application/json"
-                        })
-                        .unwrap_or(false)
-                });
+                use http::header;
 
-                let state = State::Body(body.collect());
+                if let Some(value) = ctx.request().headers().get(header::CONTENT_TYPE) {
+                    match value.as_bytes() {
```
[the remark this answers is not in the document]
Comment (carllerche) [reply src/extract/serde.rs]: Assign the lower cased version to a variable and match on that.

`let content_type = value.as_bytes().to_ascii_lowercase()`

File: src/extract/serde.rs (hunk 1 above)
Comment (carllerche) [reply src/extract/serde.rs]: If it compiles 🤷‍♂️ style will need to be cleaned up later anyway.

File: src/extract/serde.rs (hunk 2 above)
Comment (carllerche) [reply src/extract/serde.rs]: 👍

**Comment by EloD10** [comment]
Done! 👍

**Pull request merged**